File: src/main/java/me/cobble/packchooser/DatapackChooser.java
package me.cobble.packchooser;

import com.google.gson.JsonObject;
import me.cobble.packchooser.utils.PackManifests;
import net.fabricmc.api.ModInitializer;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class DatapackChooser implements ModInitializer {

    private static JsonObject manifest;
    private static final String MOD_ID = "packchooser";
    private static final Logger LOGGER = LoggerFactory.getLogger(getModId());

    @Override
    public void onInitialize() {
        manifest = PackManifests.getManifests();
    }

    public static JsonObject getManifest() {
        return manifest;
    }

    public static String getModId() {
        return MOD_ID;
    }

    public static Logger getLogger() { return LOGGER; }
}

File: src/main/java/net/datapackhub/datapacker/DatapackChooser.kt
package net.datapackhub.datapacker

import net.datapackhub.datapacker.utils.Manifests
import net.fabricmc.api.ModInitializer
import org.slf4j.LoggerFactory

object DatapackChooser : ModInitializer {

    private const val MOD_ID = "packchooser"
    private val logger = LoggerFactory.getLogger(MOD_ID)

    //    private val configPath = FabricLoader.getInstance().configDir.resolve(getModId())
//    private val config = GsonConfigInstance(Config::class.java, configPath.resolve("config.json"))
    override fun onInitialize() {
        Manifests.downloadManifest()
    }

    fun getLogger() = logger

    fun getModId() = MOD_ID

}